qa/workunits/ceph-helpers.sh

```diff
@@ -146,6 +146,27 @@ function test_teardown() {
 #######################################################################
 
 ##
+# Kills a single daemon. This is an helper function fo kill_daemons
+#
+function kill_daemon() {
+    local pid=$(cat $1)
+    local send_signal=$2
+    local delays=${4:-0.1 0.2 1 1 1 2 3 5 5 5 10 10 20 60 60 60 120}
+    local exit_code=1
+    for try in $delays ; do
+         if kill -$send_signal $pid 2> /dev/null ; then
+            exit_code=1
+         else
+	    exit_code=0
+            break
+         fi
+         send_signal=0
+         sleep $try
+    done;
+    return $exit_code
+}
+
+##
```

Comment:
You can create the test function by moving some of the test cases from test_kill_daemons. The important point here is that someone maintaining this function should not need to look anywhere else to fix a bug or add a feature. It should be self contained. Sure, the kill_deamons tests also cover kill_daemon. But when a newcomer works on kill_daemon, it is more work for her/him to figure that out. 

This helper file has been carefully designed and maintained to preserve its maintainability and usability.